ARCHITECTURE.md

```diff
@@ -0,0 +1,226 @@
+to an IstioControlPlane CR.
+1. [Manifest creation](Manifest_creation) code. Manifest rendering is done by overlaying user settings on top of a
+selected profile and passing the resulting values to a renderer in the Helm library to create manifests. Further
+overlays on the created manifests may be applied if they are defined in the user settings.
+1. [CLI](CLI) code. CLI code shares the IstioControlPlane API with the controller, but allows manifests to be generated
+and optionally applied from the command line without the need to run a privileged controller in the cluster.
+1. [Migration tools](Migration_tools). The migration tools are intended to automate configuration migration from Helm to
+the operator.
+
+The operator code uses the new Helm charts in the [istio/installer](https://github.com/istio/installer) repo. It is not
+compatible with the older charts in [istio/istio](https://github.com/istio/istio/tree/master/install/kubernetes/helm).
+See the istio/installer repo for details about the new charts and why they were created. Briefly, the new charts
+are intended to support more realistic production deployments of Istio and hitless upgrades with canarying.
+
+## Terminology
+
+Throughout the document, the following terms are used:
+
+- New API: The API directly defined in the
+[IstioControlPlane proto](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto),
+including feature and component groupings, namespaces and enablement, and per-component K8s settings. The term new API
+in this document excludes the pass-through to Helm values fields in the proto.
+- Old API: The Helm values.yaml API, implicitly defined through the various values.yaml files in the
+[Helm charts](https://github.com/istio/installer) and schematized in the operator through
+[Go structs](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/values_types.go).
+
+## The new API
+
+The new API is intended to replace the installation and K8s parts of Helm values.yaml.
+
+### Features and components
+
+The operator has a very similar structure to istio/installer: components are grouped into features.
+The operator API defines functional settings at the feature level. Functional settings are those that performs some
+function in the Istio control plane without necessarily being tied to any one component that runs in a Deployment.
+Component settings are those that necessarily refer to a particular Deployment or Service. For example, the number
+of Pilot replicas is clearly a component setting, because it refers to a component which is a Deployment in the
+cluster. Most K8s platform settings are necessarily component settings.
+The available features and components are as follows:
+
+| Feature | Components |
+|---------|------------|
+Base | CRDs
+Traffic Management | Pilot
+Policy | Policy
+Telemetry | Telemetry
+Security | Citadel
+| | Node agent
```

Comment:
are these not implemented?  if so, ight add a third field which is there state (Not implemented, alpha, beta, stable).

Reply:
They are, I think the table layout was just not clear. Updated.